hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -168,7 +169,17 @@ class IncrementalRelation(val sqlContext: SQLContext,
       } else {
         log.info("Additional Filters to be applied to incremental source are :" + filters.mkString("Array(", ", ", ")"))
 
-        var df: DataFrame = sqlContext.createDataFrame(sqlContext.sparkContext.emptyRDD[Row], usedSchema)
+        var prunedSchema = StructType(Seq())
+        if (!requiredColumns.contains(HoodieRecord.COMMIT_TIME_METADATA_FIELD)) {
```

Comment:
Add a code comment here to explain why COMMIT_TIME_METADATA_FIELD is required in prunedSchema even if it is not requested by user and not present in requiredColumns (because it is required to match the incremental fetch).

Reply:
Ack.